### PR TITLE
staging-next: assorted Python fixes and updates

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Fixes things that were still broken after the azure-core update (for mostly unrelated reasons).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/apache-airflow/default.nix [hunk 1]
```diff
@@ -243,5 +243,6 @@ buildPythonPackage rec {
     homepage = "https://airflow.apache.org/";
     license = licenses.asl20;
     maintainers = with maintainers; [ bhipple costrouc ingenieroariel ];
+    broken = true;
```
Comment (mweinelt) [inline pkgs/development/python-modules/apache-airflow/default.nix:246]: Add a comment why it is broken.

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/apache-airflow/default.nix (hunk 1 above)
Comment (K900) [reply pkgs/development/python-modules/apache-airflow/default.nix:246]: Done.

**Review by cpcloud — APPROVED** [review]
Thanks for doing this!

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 173251` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>48 packages built:</summary>
  <ul>
    <li>hydrus</li>
    <li>mailman-web (python39Packages.mailman-web)</li>
    <li>netbox</li>
    <li>paperless-ngx</li>
    <li>python310Packages.django-graphiql-debug-toolbar</li>
    <li>python310Packages.django-modelcluster</li>
    <li>python310Packages.django-oauth-toolkit</li>
    <li>python310Packages.django-rest-auth</li>
    <li>python310Packages.django-taggit</li>
    <li>python310Packages.django-timezone-field</li>
    <li>python310Packages.djangorestframework</li>
    <li>python310Packages.djangorestframework-simplejwt</li>
    <li>python310Packages.drf-jwt</li>
    <li>python310Packages.drf-nested-routers</li>
    <li>python310Packages.drf-yasg</li>
    <li>python310Packages.graphene-django</li>
    <li>python310Packages.hyperkitty</li>
    <li>python310Packages.jupyterlab</li>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-lsp</li>
    <li>python310Packages.jupyterlab_server</li>
    <li>python310Packages.jupytext</li>
    <li>python310Packages.mailman-web</li>
    <li>python310Packages.mkdocs-material</li>
    <li>python310Packages.openapi-core</li>
    <li>python310Packages.pymdown-extensions</li>
    <li>python39Packages.django-graphiql-debug-toolbar</li>
    <li>python39Packages.django-modelcluster</li>
    <li>python39Packages.django-oauth-toolkit</li>
    <li>python39Packages.django-rest-auth</li>
    <li>python39Packages.django-taggit</li>
    <li>python39Packages.django-timezone-field</li>
    <li>python39Packages.djangorestframework</li>
    <li>python39Packages.djangorestframework-simplejwt</li>
    <li>python39Packages.drf-jwt</li>
    <li>python39Packages.drf-nested-routers</li>
    <li>python39Packages.drf-yasg</li>
    <li>python39Packages.graphene-django</li>
    <li>python39Packages.hyperkitty</li>
    <li>python39Packages.jupyterlab</li>
    <li>python39Packages.jupyterlab-git</li>
    <li>python39Packages.jupyterlab-lsp</li>
    <li>python39Packages.jupyterlab_server</li>
    <li>python39Packages.jupytext</li>
    <li>python39Packages.mkdocs-material</li>
    <li>python39Packages.openapi-core</li>
    <li>python39Packages.pymdown-extensions</li>
    <li>seahub</li>
  </ul>
</details>

**Comment by K900** [comment]
Noted.

**Pull request merged**